src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -121,24 +175,40 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
             result.setTree(tree);
         }
         catch (ParseCancellationException | IllegalArgumentException ignored) {
-            // If syntax error occurs then message is printed by error listener
+            // If syntax error occurs then error listeners are accordingly notified
             // and parser throws this runtime exception to stop parsing.
             // Just stop processing current Javadoc comment.
-            final ParseErrorMessage parseErrorMessage = errorListener.getErrorMessage();
-            result.setParseErrorMessage(parseErrorMessage);
         }
 
+        // Error listener is also notified of non-tight HTML in javadoc
+        // if errorListenerOption is true
+        result.setParseErrorMessage(errorListener.getErrorMessage());
+
         return result;
     }
 
     /**
      * Parses block comment content as javadoc comment.
      * @param blockComment
      *        block comment content.
+     * @param errorListenerOption Supplied by
+     *     {@link #parseJavadocAsDetailNode(DetailAST, ErrorListenerOption)} when it calls this
+     *     method. The javadoc grammar is designed in such a way to enable the JavadocParser to
+     *     store the first non-Tight HTML tag it encounters. If the javadoc being parsed indeed
+     *     has non-tight HTML tag(s) then the error listeners attached to ANTLR parser shall
+     *     accordingly be notified of the first non-tight HTML tag encountered provided
+     *     {@link ErrorListenerOption#NOTIFY_FOR_NON_TIGHT_HTML} is the argument supplied.
      * @return parse tree
-     * @noinspection deprecation
```

Comment:
Then why was check for this PR was failing. Can you please have a look at https://teamcity.jetbrains.com/viewLog.html?buildId=1144804&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest and tell me what is wrong exatly ?

Also commenting just the statement for `noinspection`  seems better than `noinspection` for the entire method, doesn't it ? Assuming that the comment over the statement doesn't apply to the whole method anyway.